### PR TITLE
Fixed multiple bugs in the demo tool

### LINKED ISSUE #155: Add new data providers to the demo tool
Several data providers were introduced in 0.6.0. However, we forgot to enable them in the demo tool.

### LINKED ISSUE #156: UsesGithubForDevelopment fails with UnsupportedTemporalTypeException
https://github.com/SAP/fosstars-rating-core/blob/master/src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesGithubForDevelopment.java#L107

Looks like `MONTHS` is not supported. Let's update it to `DAYS`.

### LINKED ISSUE #157: UsesDependabot.isDependabot() fails with NPE
Here is a stack trace:

```
java.lang.NullPointerException
	at com.sap.sgs.phosphor.fosstars.data.github.UsesDependabot.isDependabot(UsesDependabot.java:137)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesDependabot.hasDependabotCommits(UsesDependabot.java:80)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesDependabot.usesDependabot(UsesDependabot.java:132)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesDependabot.fetchValueFor(UsesDependabot.java:68)
	at com.sap.sgs.phosphor.fosstars.data.github.CachedSingleFeatureGitHubDataProvider.fetchValuesFor(CachedSingleFeatureGitHubDataProvider.java:29)
	at com.sap.sgs.phosphor.fosstars.data.github.CachedSingleFeatureGitHubDataProvider.fetchValuesFor(CachedSingleFeatureGitHubDataProvider.java:16)
	at com.sap.sgs.phosphor.fosstars.data.AbstractCachingDataProvider.doUpdate(AbstractCachingDataProvider.java:64)
	at com.sap.sgs.phosphor.fosstars.data.AbstractCachingDataProvider.doUpdate(AbstractCachingDataProvider.java:21)
	at com.sap.sgs.phosphor.fosstars.data.AbstractDataProvider.update(AbstractDataProvider.java:51)
	at com.sap.sgs.phosphor.fosstars.data.AbstractDataProvider.update(AbstractDataProvider.java:17)
	at com.sap.sgs.phosphor.fosstars.tool.github.SingleSecurityRatingCalculator.calculateFor(SingleSecurityRatingCalculator.java:72)
	at com.sap.sgs.phosphor.fosstars.tool.github.SingleSecurityRatingCalculator.calculateFor(SingleSecurityRatingCalculator.java:42)
	at com.sap.sgs.phosphor.fosstars.tool.github.MultipleSecurityRatingsCalculator.calculateFor(MultipleSecurityRatingsCalculator.java:57)
	at com.sap.sgs.phosphor.fosstars.tool.github.MultipleSecurityRatingsCalculator.calculateFor(MultipleSecurityRatingsCalculator.java:75)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.processConfig(SecurityRatingCalculator.java:246)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.run(SecurityRatingCalculator.java:174)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.main(SecurityRatingCalculator.java:93)
```

Looks like we need to add a couple of null checks.

### LINKED ISSUE #158: The demo tool should store project cache more often
When the demo tool calculates rating for many projects, it stores the project cache only after it finishes calculating all ratings. It may be better to store the cache sometimes during calculating ratings. That would help to lower the risk of losing the data on long runs.

### LINKED ISSUE #160: UsesGithubForDevelopment may fail with NPE
Looks like the `UsesGithubForDevelopment` data provider needs a couple of additional null-checks:

```
java.lang.NullPointerException
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.lambda$static$1(UsesGithubForDevelopment.java:43)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.lambda$usesGitHubForDevelopment$10(UsesGithubForDevelopment.java:132)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:474)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.usesGitHubForDevelopment(UsesGithubForDevelopment.java:133)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.usesGithubForDevelopment(UsesGithubForDevelopment.java:111)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.fetchValueFor(UsesGithubForDevelopment.java:94)
	at com.sap.sgs.phosphor.fosstars.data.github.CachedSingleFeatureGitHubDataProvider.fetchValuesFor(CachedSingleFeatureGitHubDataProvider.java:29)
	at com.sap.sgs.phosphor.fosstars.data.github.CachedSingleFeatureGitHubDataProvider.fetchValuesFor(CachedSingleFeatureGitHubDataProvider.java:16)
	at com.sap.sgs.phosphor.fosstars.data.AbstractCachingDataProvider.doUpdate(AbstractCachingDataProvider.java:64)
	at com.sap.sgs.phosphor.fosstars.data.AbstractCachingDataProvider.doUpdate(AbstractCachingDataProvider.java:21)
	at com.sap.sgs.phosphor.fosstars.data.AbstractDataProvider.update(AbstractDataProvider.java:51)
	at com.sap.sgs.phosphor.fosstars.data.AbstractDataProvider.update(AbstractDataProvider.java:17)
	at com.sap.sgs.phosphor.fosstars.tool.github.SingleSecurityRatingCalculator.calculateFor(SingleSecurityRatingCalculator.java:72)
	at com.sap.sgs.phosphor.fosstars.tool.github.SingleSecurityRatingCalculator.calculateFor(SingleSecurityRatingCalculator.java:42)
	at com.sap.sgs.phosphor.fosstars.tool.github.MultipleSecurityRatingsCalculator.calculateFor(MultipleSecurityRatingsCalculator.java:57)
	at com.sap.sgs.phosphor.fosstars.tool.github.MultipleSecurityRatingsCalculator.calculateFor(MultipleSecurityRatingsCalculator.java:75)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.processConfig(SecurityRatingCalculator.java:246)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.run(SecurityRatingCalculator.java:174)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.main(SecurityRatingCalculator.java:93)
```

### LINKED ISSUE #161: Data providers don't expect empty projects on GitHub
While calculating security ratings with the demo tool, data providers sometimes fail with the following exception. It happens when a repository is empty. It may be better to make the data providers or the demo tool aware of such cases. For example, the tool can run some smoke tests for a project before passing it to the data providers.

```
[!] Couldn't fetch data, something went wrong!
org.kohsuke.github.HttpException: {"message":"Git Repository is empty.","documentation_url":"https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository"}
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:440)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:368)
	at org.kohsuke.github.GitHubPageIterator.fetch(GitHubPageIterator.java:144)
	at org.kohsuke.github.GitHubPageIterator.hasNext(GitHubPageIterator.java:91)
	at org.kohsuke.github.PagedIterator.fetch(PagedIterator.java:106)
	at org.kohsuke.github.PagedIterator.nextPageArray(PagedIterator.java:134)
	at org.kohsuke.github.PagedIterable.toArray(PagedIterable.java:79)
	at org.kohsuke.github.PagedIterable.toArray(PagedIterable.java:107)
	at org.kohsuke.github.PagedIterable.toList(PagedIterable.java:119)
	at com.sap.sgs.phosphor.fosstars.data.github.GitHubDataFetcher.commitsFor(GitHubDataFetcher.java:97)
	at com.sap.sgs.phosphor.fosstars.data.github.GitHubDataFetcher.commitsFor(GitHubDataFetcher.java:83)
	at com.sap.sgs.phosphor.fosstars.data.github.GitHubDataFetcher.commitsAfter(GitHubDataFetcher.java:150)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.usesGithubForDevelopment(UsesGithubForDevelopment.java:108)
	at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.fetchValueFor(UsesGithubForDevelopment.java:94)
	at com.sap.sgs.phosphor.fosstars.data.github.CachedSingleFeatureGitHubDataProvider.fetchValuesFor(CachedSingleFeatureGitHubDataProvider.java:29)
	at com.sap.sgs.phosphor.fosstars.data.github.CachedSingleFeatureGitHubDataProvider.fetchValuesFor(CachedSingleFeatureGitHubDataProvider.java:16)
	at com.sap.sgs.phosphor.fosstars.data.AbstractCachingDataProvider.doUpdate(AbstractCachingDataProvider.java:64)
	at com.sap.sgs.phosphor.fosstars.data.AbstractCachingDataProvider.doUpdate(AbstractCachingDataProvider.java:21)
	at com.sap.sgs.phosphor.fosstars.data.AbstractDataProvider.update(AbstractDataProvider.java:51)
	at com.sap.sgs.phosphor.fosstars.data.AbstractDataProvider.update(AbstractDataProvider.java:17)
	at com.sap.sgs.phosphor.fosstars.tool.github.SingleSecurityRatingCalculator.calculateFor(SingleSecurityRatingCalculator.java:72)
	at com.sap.sgs.phosphor.fosstars.tool.github.SingleSecurityRatingCalculator.calculateFor(SingleSecurityRatingCalculator.java:42)
	at com.sap.sgs.phosphor.fosstars.tool.github.MultipleSecurityRatingsCalculator.calculateFor(MultipleSecurityRatingsCalculator.java:57)
	at com.sap.sgs.phosphor.fosstars.tool.github.MultipleSecurityRatingsCalculator.calculateFor(MultipleSecurityRatingsCalculator.java:75)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.processConfig(SecurityRatingCalculator.java:246)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.run(SecurityRatingCalculator.java:174)
	at com.sap.sgs.phosphor.fosstars.tool.github.SecurityRatingCalculator.main(SecurityRatingCalculator.java:93)
Caused by: java.io.IOException: Server returned HTTP response code: 409 for URL: https://api.github.com/repos/eclipse/adore/commits
	at sun.reflect.GeneratedConstructorAccessor131.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1944)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1939)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1938)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1508)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient$HttpURLConnectionResponseInfo.bodyStream(GitHubHttpUrlConnectionClient.java:197)
	at org.kohsuke.github.GitHubResponse$ResponseInfo.getBodyAsString(GitHubResponse.java:313)
	at org.kohsuke.github.GitHubResponse.parseBody(GitHubResponse.java:80)
	at org.kohsuke.github.GitHubPageIterator.lambda$fetch$0(GitHubPageIterator.java:145)
	at org.kohsuke.github.GitHubClient.createResponse(GitHubClient.java:406)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:360)
	... 25 more
Caused by: java.io.IOException: Server returned HTTP response code: 409 for URL: https://api.github.com/repos/eclipse/adore/commits
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1894)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:347)
	at org.kohsuke.github.GitHubHttpUrlConnectionClient.getResponseInfo(GitHubHttpUrlConnectionClient.java:69)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:348)
	... 25 more
```

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Fixes multiple bugs in the data gathering procedures.

This fixes #155
This fixes #156 
This fixes #157
This fixes #158
This fixes #160 
This fixes #161

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java [hunk 1]
```diff
@@ -94,7 +104,12 @@ public static GitHubDataFetcher instance() {
    * @throws IOException occurred during REST call to GitHub API.
    */
   private List<GHCommit> commitsFor(GHRepository repository) throws IOException {
-    return repository.listCommits().toList();
+    try {
+      return repository.listCommits().toList();
+    } catch (HttpException e) {
+      LOGGER.error(String.format("Could not fetch commits from %s", repository.getUrl()), e);
+      return Collections.emptyList();
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java:111]: Should this throw an IOException? after catching HttpException?

File: src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/SingleSecurityRatingCalculator.java [hunk 2]
```diff
@@ -49,11 +54,20 @@
   public SingleSecurityRatingCalculator calculateFor(GitHubProject project) throws IOException {
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/SingleSecurityRatingCalculator.java:54]: Would this function throw an exception if there is a try catch block? Is this `throws IOException` required?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java (hunk 1 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java:111]: If a repository is empty, `listCommits()` throws an exception. If it's called in a data provider, then it fails. So, there should be a check somewhere. I don't think that returning an empty list makes any harm at the moment.

File: src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/SingleSecurityRatingCalculator.java (hunk 2 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/SingleSecurityRatingCalculator.java:54]: Yes, the method can still throw an `IOException`.

**Comment by artem-smotrakov** [comment]
Thanks for the review @sourabhsparkala !

**Pull request merged**